src/relay/transforms/simplify_expr.cc

```diff
@@ -585,6 +585,47 @@ class EliminateIdentityRewrite : public DFPatternRewrite {
   DFPattern const_;
 };
 
+/*! \brief Make two consecutive add able to be constant_folded. */
+class SimplifyConsecutiveAdd : public DFPatternRewrite {
+ public:
+  SimplifyConsecutiveAdd() {
+    x_ = IsWildcard();
+    const1_ = IsConstant();
+    const2_ = IsConstant();
+    DFPattern add_op1_ = IsOp("add");
+    DFPattern add_op2_ = IsOp("add");
+    pattern_ = add_op2_({add_op1_({x_, const1_}), const2_});
+  }
+
+  Expr Callback(const Expr& pre, const Expr& post,
+                const Map<DFPattern, Array<Expr>>& node_map) const override {
+    const CallNode* call = pre.as<CallNode>();
+    auto x = node_map[x_][0];
+    auto c1 = node_map[const1_][0];
+    auto c2 = node_map[const2_][0];
+
+    auto pre_call = call;
+    if (pre_call->args[1].as<ConstantNode>()) {
+      pre_call = pre_call->args[0].as<CallNode>();
+    } else {
+      pre_call = pre_call->args[1].as<CallNode>();
+    }
+    // check the first two input has one non-constant
```

Comment:
```suggestion
    // Do nothing if both inputs are not constants as they will be constant folded already.
```

Reply:
Thank you for your comment.
1. Unnecessary underline suffix has been removed.
2. Unnecessary `add_op` has been removed.
3. A comment saying `This pattern matching supports commutative property for addition.` has been added.